### DISCUSSION TITLE
OSC Customization - How to simply move Background

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14871] [General]
Linux Debian based distro
MPV 0.38

The OSC (osc.lua file) has a default black box background.
```
local function bar_layout(direction)
    local osc_geo = {
        x = -2,
        y = nil,
        an = (direction < 0) and 7 or 1,
        w = nil,
        h = 56,
    }
```
I was able to figure out how to shorten the height by changing the "h = 56” to desired height I was looking for.
When you do this the height adjusts from the bottom up which means now need to move the background down which at the moment I cannot figure out how to do.
Does anyone know how to move this background down?
Thank You

`SOLVED`
I was able to figure this out if anyone in the future is interested in this.
The "h = #" value mentioned above also has to be changed in the following line which you should see below the code mentioned above
```
osc_geo.y = direction * (56 + user_opts.barmargin)  --THIS VALUE HERE CURRENTLY SET AT 56
osc_geo.w = osc_param.playresx + 4
```
Hopefully someone will find this useful in the future.
Thanks for the help

**Comment by advice2020** [comment]
Just keep in mind that when you do this it leads to other issues because everything in the OSC is linked to each other.
So then items like timecodes, seekbar, etc. will then move down with it getting cut off by the bottom of your screen, which I have not figured out how to fix yet.

**Comment by advice2020** [comment]
@Samillion
Here is the post I was mentioning if you are interested, you may not have seen this OSC based post because I had closed it and now reopened it.
This was one of the main reasons for my "geo" question post, how to actually move items.
Once I changed the h = 56 height value to a shorter height all the items based off of it shifted down with it :)
So I was trying to figure out how to move them back up.

If the timecodes and seekbar for example are using "y = geo.y" already, then I would imagine that everything should just adjust properly and position itself based on the new background height, but they do not.
If anyone has any ideas.

**Comment by JorgeR81** [comment]
> Just keep in mind that when you do this it leads to other issues because everything in the OSC is linked to each other.
> So then items like timecodes, seekbar, etc. will then move down with it getting cut off by the bottom of your screen, which I have not figured out how to fix yet.

@advice2020, did you fully solved your issue ?

To have a lower value instead of `h=56`, I decreased the value in `line 2`, by the same amount.

For instance, to set `h=36`, you need change:

 `local line2 = osc_geo.y - direction * (36 + padY) ` 

to:
 `local line2 = osc_geo.y - direction * (16 + padY) `


---

I didn't change the `osc_geo.y` value.

I did this a long time ago in the mpv `0.32.0` OSC, but in still works the same way in `0.40.0`

![slimbar](https://github.com/user-attachments/assets/061b9d3f-93e1-4b81-82c0-1af17c477290)
